pkg/base/replicated.go

```diff
@@ -206,24 +266,27 @@ func renderReplicatedHelmChart(kotsHelmChart *kotsv1beta1.HelmChart, upstreamFil
 	if mergedValues == nil {
 		mergedValues = map[string]kotsv1beta1.MappedChartValue{}
 	}
-	for _, optionalValues := range kotsHelmChart.Spec.OptionalValues {
-		parsedBool, err := strconv.ParseBool(optionalValues.When)
+
+	for _, optionalValue := range kotsHelmChart.Spec.OptionalValues {
+		parsedBool, err := strconv.ParseBool(optionalValue.When)
 		if err != nil {
-			return nil, errors.Wrap(err, "failed to parse when conditional on optional values")
+			return nil, errors.Wrap(err, "failed to parse when conditional on optional value")
 		}
+
 		if !parsedBool {
 			continue
 		}
-		if optionalValues.RecursiveMerge {
-			mergedValues = kotsv1beta1.MergeHelmChartValues(mergedValues, optionalValues.Values)
+
+		if optionalValue.RecursiveMerge {
+			mergedValues = kotsv1beta1.MergeHelmChartValues(mergedValues, optionalValue.Values)
 		} else {
-			for k, v := range optionalValues.Values {
+			for k, v := range optionalValue.Values {
 				mergedValues[k] = v
 			}
 		}
 	}
 
-	helmValues, err := kotsHelmChart.Spec.GetHelmValues(mergedValues)
+	helmValues, err := kotsHelmChart.Spec.GetHelmValues(kotsHelmChart.Spec.Values)
```

Comment:
is this right? shouldn't this be:
```suggestion
	helmValues, err := kotsHelmChart.Spec.GetHelmValues(mergedValues)
```

Reply:
i'm worried if all tests pass with the current change

Reply:
I only added expected kotskinds to the existing tests and didn't modify any of the cases. It looks like there aren't any tests that use HelmChart.Spec.OptionalValues, `rg OptionalValue pkg/tests` returns nothing.